src/core/master/src/main/java/org/drftpd/master/network/BaseFtpConnection.java

```diff
@@ -454,13 +453,18 @@ public void run() {
                 }
 
                 // execute command
-                _pool.execute(new CommandThread(_request, this));
-                if (_request.getCommand().equalsIgnoreCase("AUTH")) {
-                    while (!_securityDataExchangeCompleted && !_stopRequest) {
-                        Thread.sleep(100);
+                // If we get ABOR handle it directly otherwise hand it over to the executor
+                if (_request.getCommand().equalsIgnoreCase("ABOR")) {
```

Comment:
Not sure how it works but for me every user have a thread of BaseFtpConnection. Inside this thread I dont think a user can a multiple commands execution at the same time. So maybe you just have to remove the pool and simplify your code.

Reply:
I agree, but so no way to actually handle the "ABOR" use case per above without doing it this way. I am open to suggestions tho